File: example-criteria/src/test/java/example/criteria/BatchUpdateTest.java
package example.criteria;

import example.common.domain.Salary;
import example.common.entity.Employee;
import example.common.test.TestEnvironment;
import example.criteria.repository.EmployeeRepository;
import java.util.List;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.seasar.doma.jdbc.Config;

@ExtendWith(TestEnvironment.class)
public class BatchUpdateTest {

  private final EmployeeRepository repository;

  BatchUpdateTest(Config config) {
    repository = new EmployeeRepository(config);
  }

  @Test
  public void testBatchUpdate() {
    List<Employee> list = repository.selectAll();
    for (Employee employee : list) {
      Salary salary = employee.getSalary();
      if (salary != null) {
        employee.setSalary(salary.add(new Salary(100)));
      }
    }
    repository.batchUpdate(list);
  }
}

File: dsl-style-kotlin/src/test/kotlin/example/dsl_style_kotlin/BatchUpdateTest.kt
package example.dsl_style_kotlin

import example.dsl_style_kotlin.domain.Salary
import example.dsl_style_kotlin.repository.EmployeeRepository
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.seasar.doma.jdbc.Config

@ExtendWith(TestEnvironment::class)
class BatchUpdateTest(config: Config) {

    private val repository: EmployeeRepository = EmployeeRepository(config)

    @Test
    fun testBatchUpdate() {
        val list = repository.selectAll()
        for (employee in list) {
            val salary = employee.salary
            if (salary != null) {
                employee.salary = salary.add(Salary(100))
            }
        }
        repository.batchUpdate(list)
    }
}
